File: app-java/app/src/main/java/com/phunware/android/phunwareproducthomework/dagger/ZipCodeStoreModule.java
package com.phunware.android.phunwareproducthomework.dagger;

import com.phunware.android.phunwareproducthomework.storage.InMemoryZipCodeStore;
import com.phunware.android.phunwareproducthomework.storage.ZipCodeStore;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ZipCodeStoreModule {
    @Provides
    @Singleton
    ZipCodeStore provideZipCodeStore() {
        return new InMemoryZipCodeStore();
    }
}

File: app-kotlin/app/src/main/java/com/phunware/android/phunwareproducthomework/dagger/ZipCodeStoreModule.kt
package com.phunware.android.phunwareproducthomework.dagger

import com.phunware.android.phunwareproducthomework.storage.InMemoryZipCodeStore
import com.phunware.android.phunwareproducthomework.storage.ZipCodeStore
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ZipCodeStoreModule {
  @Provides
  @Singleton
  fun provideZipCodeStore(): ZipCodeStore = InMemoryZipCodeStore()
}
